src/OnChain.ts

```diff
@@ -136,62 +138,100 @@ export const OnChainMixin = (superclass) => class extends superclass {
         throw new TransactionRestrictedError(error,{logger: onchainLogger})
       }
 
+      // TODO? should onChainBalance also be checked by itself when isSendAll? (was originally checked after adding the estimatedFee)
```

Comment:
yes I think so.

this is to make sure we have enough money in the hot wallet. if not we fail early. this is independant of sendAll

(in a future version we should have a queue here so we record the transaction and when for rebalancing from the cold storage)